src/app/seller-post-ad-form/seller-post-ad-form.component.ts

```diff
@@ -0,0 +1,181 @@
+  imageArray:ImageProperty[] =[];
+  
+  categories = ["Home","Electronics","Car","Bike"]; // this is provided by categories api
+  states = ["Andra Pradesh","Go","Gujarat","Haryana","Himachal Pradesh","Jammu and Kashmir","Jharkhand","Karnataka",
+  "Kerala","Madya Pradesh","Maharashtra","Punjab","Rajasthan"]
+  
+  //properties of html element 
+  addressDisplayValue = "none";
+  purchaseDate = "none";
+  imageCounter = 1;
+
+  constructor(public datepipe: DatePipe){} //use for validation of date 
+
+  ngOnInit() {
+    let image =new ImageProperty();
+    this.imageArray.push(image);
+  } 
+
+  getImageUrlArray() 
+  {
+    for (let index = 0; index < this.imageArray.length-1; index++) {
+      this.imageUrlArray.push(this.imageArray[index].imageURL);
+    }
+    return this.imageUrlArray;
+  }
+  
+  onPost(postAdForm : NgForm)
+  {
+    for (let index = 0; index < this.imageArray.length-1; index++) {
+      this.imageUrlArray.push(this.imageArray[index].imageURL);
+    }
+    console.log(postAdForm.value);
+    console.log(JSON.stringify(postAdForm.value));
+  }
+
+  date =  new Date();
+  validateDate(id)
+  {
+    this.date=new Date();
+    let latest_date =this.datepipe.transform(this.date, 'yyyy-MM-dd');
+    var userDate = id.target.value;
+    if(userDate > latest_date){
+      this.purchaseDate="block";
+    }else{
+      this.purchaseDate="none";
+    }
+  }
+
+  validateAddress(event)
+  {
+    if(event.target.checked)
+    {
+      this.addressDisplayValue="block";
+      this.isAddressSelected=true;
+    }
+    else{
+      this.addressDisplayValue="none"
+      this.isAddressSelected=false;
+    }
+  }
+
+  imageLoader(id)
+  {
+    this.imageArray[id].CrossBtnValue="none";
+    this.imageArray[id].imageDisplayValue="none";
+    this.imageArray[id].addEditProperty="none";
+    this.imageArray[id].pictureContainerStyle = "1px solid lightgrey";
+    this.imageArray[id].heroImage = "none";
+    this.imageArray[id].imageLoaderProperty="";
+  }
+
+  async delay(ms: number,id) {
+    await new Promise(resolve => setTimeout(()=>resolve(), ms)).then(()=>console.log("image Loading"));
+    this.imageArray[id].addEditProperty="";
+    this.imageArray[id].CrossBtnValue="";
+    this.imageArray[id].imageDisplayValue="";
+    this.imageArray[id].imageURL = "https://images.pexels.com/photos/170811/pexels-photo-170811.jpeg?auto=compress&cs=tinysrgb&dpr=1&w=500";
+    this.imageArray[id].buttonName ="Change";
+    this.imageArray[id].iconOfButton = "edit";
+    this.imageArray[id].imageLoaderProperty="none";
+    if(id==0)
+    {
+      this.selectHeroImg(id);
+      this.imageArray[0].heroImage="";
+    }
+  }
+  
+  addImage(id,event){
+    if(id==0)
+    {
+      this.selectHeroImg(id);
+      this.imageArray[0].heroImage="";
+    }
+    
+    this.imageLoader(id);
+    this.delay(3000,id);
+    console.log(event.target.value);  // calls upload image api it will return url of image after uploading image
```

Comment:
remove extra comments that are not required

Reply:
done